ocf_data_sampler/load/satellite.py

```diff
@@ -1,59 +1,230 @@
+        except Exception as e:
+            # NO FALLBACK - error out as Sol requested
+            raise ValueError(f"Commit {sha1} not found or inaccessible: {e}")
+    else:
+        logger.info("Opening 'main' branch of Ice Chunk repository.")
+        session = repo.readonly_session("main")
+
+    # Open the dataset from the Ice Chunk session store
+    ds = xr.open_zarr(session.store, consolidated=True, chunks="auto")
 
-    check_time_unique_increasing(ds.time_utc)
+    # Convert Ice Chunk format to standard format
+    if len(ds.data_vars) > 1:
+        # Data is stored with separate variables per channel - combine them
+        data_arrays = [ds[var] for var in sorted(ds.data_vars)]
+        combined_da = xr.concat(data_arrays, dim="variable")
+        combined_da = combined_da.assign_coords(variable=sorted(ds.data_vars))
+        
+        # Create dataset with data variable (to match OCF format)
+        ds = xr.Dataset({"data": combined_da})
+
+    return ds
+
+def get_single_sat_data(zarr_path: str) -> xr.Dataset:
+    """Opens a single satellite zarr file with Ice Chunk support via match/case patterns.
+    
+    This function implements Sol's requested architecture consolidating all conditional 
+    logic into match/case patterns for different path types.
+    """
+    # Parse path components using Sol's regex approach
+    path_info = _parse_icechunk_path(zarr_path)
+    
+    # Sol's requested match/case pattern for path routing
+    match path_info:
+        case {"protocol": None, "bucket": _, "prefix": _, "sha1": sha1} if sha1 is not None:
+            # Raise error if trying to use commit syntax for local file
+            raise ValueError("Commit syntax (@commit) not supported for local files")
+        
+        case {"protocol": "gs", "bucket": _, "prefix": prefix, "sha1": _} if "*" in prefix:
+            # Raise an error if a wildcard is used in a GCP path
+            raise ValueError("Wildcard (*) paths are not supported for GCP (gs://) URLs")
+        
+        case {"protocol": protocol, "bucket": bucket, "prefix": prefix, "sha1": sha1} if ".icechunk" in prefix and protocol is not None:
+            # Ice Chunk logic goes here - Sol's requested signature
+            ds = _open_sat_data_icechunk(protocol, bucket, prefix, sha1)
+        
+        case {"protocol": _, "bucket": _, "prefix": _, "sha1": None}:
+            # Existing single zarr logic path here - Sol's specification
+            ds = xr.open_dataset(zarr_path, engine="zarr", chunks="auto")
+        
+        case {"protocol": None, "bucket": _, "prefix": prefix, "sha1": None} if "*" in prefix:
+            # Handle multi-zarr dataset for local files
+            ds = xr.open_mfdataset(
+                zarr_path,
+                engine="zarr",
+                concat_dim="time",
+                combine="nested",
+                chunks="auto",
+                join="override",
+            )
+            check_time_unique_increasing(ds.time)
```

Comment:
This can go because you take care of multiple zarrs in the `open_sat_data` function